### ISSUE TITLE
you can add firedrake.vector.Vectors, but not subtract them

### ISSUE THREAD

**Issue by APaganini** [firedrakeproject/firedrake#958]
(I'm sorry but I don't know how to generate a minimal test for this without firedrake-adjoint
because I don't know how to create an instance of firedrake.vector.Vector`)

```
from firedrake import *
from firedrake_adjoint import *

mesh = UnitSquareMesh(4, 4)
V = FunctionSpace(mesh, "CG", 1)
f = project(Expression("x[0]*(x[0]-1)*x[1]*(x[1]-1)", degree=1), V)

def main(f, annotate=True):
    u = TrialFunction(V)
    v = TestFunction(V)

    u = Function(V, name="Solution")
    F = (inner(grad(u), grad(v)) + f*v)*dx
    bc = DirichletBC(V, 1.0, "on_boundary")
    solve(F == 0, u, bc, annotate=annotate)
    return u

if __name__ == "__main__":

    u = main(f)
    m = Control(f)

    J1 = Functional(u**2 * dx)
    dJ1dm = compute_gradient(J1, m, forget=False)

    (dJ1dm.vector() -  dJ1dm.vector())
```

returns the error

*** TypeError: unsupported operand type(s) for -: 'Vector' and 'Vector'

**Comment by wence-** [comment]
Can you implement `__sub__` and `__isub__` on `Vector`, following the pattern of `__add__` and `__iadd__`?

**Comment by APaganini** [comment]
I'll try

**Comment by wence-** [comment]
Fixed by #962.
